### ISSUE TITLE
Why are the style objects not cached?

### ISSUE THREAD

**Issue by nicolasfranck** [zotero/citeproc-js-server#43]
The current version does not cache the parsed csl style objects
but loads them on every request.

This is due to the following reasons:

* [NoncacheEngineCache](https://github.com/zotero/citeproc-js-server/blob/master/lib/citeServer.js#L120) is used. That implementation loads the style object from disk (See [here](https://github.com/zotero/citeproc-js-server/blob/master/lib/engineCaching.js#L47) and [here](https://github.com/zotero/citeproc-js-server/blob/4d9eb3f1ce8026bc1756aefdb0ac187c5276b298/lib/styles.js#L257)).
* It is possible to use QueueCache, but that implementation has possible data racing issues, not? I see that every "engine" has attributes like "working" (because it stores data about the thing it is working on), and that the engine is ["reset"](https://github.com/zotero/citeproc-js-server/blob/master/lib/citeServer.js#L298) before returning it to the cache. So if parallel requests come in, the same engine is used. Even [this line](https://github.com/zotero/citeproc-js-server/blob/master/lib/engineCaching.js#L131) does not really solve the issue.

Why not simply cache the parsed style objects (which are immutable)? Those are the slowest to load.
In that case there is no need for an engine cache, as that complicates matters.

In short: the caching is done at the wrong place?
